API/events/triggerEvent.m

```diff
@@ -36,22 +36,35 @@ function triggerEvent(eventType, data)
                 coder.ceval('std::mem_fn(&eventHelper::sendMessage)', helper, [data,0]);
             elseif strcmpi(eventType, 'plot')
                 result = data{1};
+                problemDef = data{3};
                 nContrast = length(result{1});
-                [reflect, nReflect] = packCellArray(result{1}); % reflectivity
-                [shiftedData, nShiftedData] = packCellArray(result{3});  
-                [sldProfiles, nSldProfiles] = packCellArray(result{5});
-                [layers, nLayers] = packCellArray(result{6}); % All Layers
-              
+                [reflect, nReflect] = packCellArray(result{1}, 1); % reflectivity
+                [shiftedData, nShiftedData] = packCellArray(result{3}, 1);  
+   
+                switch problemDef.TF
+                    case 'domains'
+                        [sldProfiles, nSldProfiles] = packCellArray(result{5}, 1);
+                        [sldProfiles2, nSldProfiles2] = packCellArray(result{5}, 2);
+                        [layers, nLayers] = packCellArray(result{6}, 1);
+                        [layers2, nLayers2] = packCellArray(result{6}, 2);
+                    otherwise
+                        [sldProfiles, nSldProfiles] = packCellArray(result{5}, 1);
+                        [layers, nLayers] = packCellArray(result{6}, 1);
```

Comment:
These two lines are also included in the previous clause. Presumably they can be moved out of the switch case?

Reply:
yes, will do